src/packages/@ncigdc/components/Modals/AppendSetModal.js

```diff
@@ -41,61 +40,74 @@ const AppendSetModal = ({
+          value: `set_id:${selected}`,
+        },
+      }}
+      {...p}
+    />,
 }) => {
   return (
     <BaseModal
       title={title}
+      closeText="Cancel"
       extraButtons={
-        <AppendSetButton
-          disabled={!selected}
-          set_id={`set_id:${selected}`}
-          filters={filters}
-          size={inputTotal}
-          sort={sort}
-          score={score}
-          onComplete={setId => {
-            if ((query.filters || '').includes(selected)) {
-              history.replace({
-                search: `?${stringify({
-                  ...query,
-                  filters: query.filters.replace(selected, setId),
-                })}`,
-              });
-            }
+        !selected
+          ? null
+          : <Count>
+              {count =>
+                <AppendSetButton
+                  disabled={!selected}
+                  set_id={`set_id:${selected}`}
+                  filters={filters}
+                  size={Math.min(max, count)}
+                  sort={sort}
+                  score={score}
+                  onComplete={setId => {
+                    if ((query.filters || '').includes(selected)) {
+                      history.replace({
+                        search: `?${stringify({
+                          ...query,
+                          filters: query.filters.replace(selected, setId),
+                        })}`,
+                      });
+                    }
 
-            onSaveComplete({
-              dispatch,
-              label: sets[selected],
-            });
+                    onSaveComplete({
+                      dispatch,
+                      label: sets[selected],
+                    });
 
-            dispatch(replaceSet({ type, oldId: selected, newId: setId }));
-          }}
-        >
-          Save
-        </AppendSetButton>
+                    dispatch(
+                      replaceSet({ type, oldId: selected, newId: setId }),
+                    );
+                  }}
+                >
+                  Save
+                </AppendSetButton>}
+            </Count>
       }
-      closeText="Cancel"
     >
-      <label>
-        Save top:<br />
-        <input
-          type="number"
-          max={max}
-          value={inputTotal}
-          onChange={e => setInputTotal(e.target.value)}
-        />
-        <div style={{ fontSize: '0.8em' }}>
-          You can append up to the top {pluralize(displayType, max, true)}
-        </div>
-      </label>
-      {inputTotal > max &&
-        <WarningBox>
-          Above maximum of {pluralize(displayType, max, true)}
-        </WarningBox>}
-
+      {selected &&
+        <Count>
+          {count =>
+            <span>
+              {total + count >= max &&
+                <WarningBox>
+                  Above maximum of {pluralize(displayType, max, true)}.
+                  {count < max &&
+                    <span>&nbsp;Only {max - count} will be saved.</span>}
```

Comment:
Didn't we talk about checking for duplicate ids? or did that not work out?

Reply:
sorry I'm a little fuzzy on the details of that conversation. You mean to make sure that only items which are not already in the set are added, right?

Reply:
more making it so it doesn't count duplicate ids against the total. 
so I believe right now if you were to save 30,000 and then append the same 30,000 it would tell you it's being truncated when really there are only 30,000 unique ids.
only really an issue if the append also contains 1 new id but then is not added because of the truncating.